guis/qml/ut/MainUt.qml

```diff
@@ -15,256 +15,290 @@ UITK.Page {
+      root.handler = ContentHandler.Source
+      root.selectionType = ContentTransfer.Single
+    }
+    else if(request.message =="document")
+    {
+      root.request = request
+      root.requestContentHub = true
+      root.contentType = ContentType.Documents
+      root.handler = ContentHandler.Source
+      root.selectionType = ContentTransfer.Single
+    }
+    else if(request.message =="audio")
+    {
+      root.request = request
+      root.requestContentHub = true
+      root.contentType = ContentType.Music
+      root.handler = ContentHandler.Source
+      root.selectionType = ContentTransfer.Single
+    }
+    else if(request.message.includes("[oC]"))
+    {
+      root.request = request
+      root.requestContentHub = true
+      root.url = request.message.substring(4)
+      root.contentType = ContentType.Contacts
+      root.handler = ContentHandler.Destination
+      root.selectionType = ContentTransfer.Multiple
+    }
+    else if(request.message.includes("[oP]"))
+    {
+      root.request = request
+      root.requestContentHub = true
+      root.url = request.message.substring(4)
+      root.contentType = ContentType.Pictures
+      root.handler = ContentHandler.Destination
+      root.selectionType = ContentTransfer.Single
+    }
+    else if(request.message.includes("[oV]"))
+    {
+      root.request = request
+      root.requestContentHub = true
+      root.url = request.message.substring(4)
+      root.contentType = ContentType.Videos
+      root.handler = ContentHandler.Destination
+      root.selectionType = ContentTransfer.Single
+    }
+    else if(request.message.includes("[oD]"))
+    {
+      root.request = request
+      root.requestContentHub = true
+      root.url = request.message.substring(4)
+      root.contentType = ContentType.Documents
+      root.handler = ContentHandler.Destination
+      root.selectionType = ContentTransfer.Single
+    }
+    else if(request.message.toLowerCase().includes("call"))
+    {
+      root.request = request
+      var callUrl = request.message.substring(4)
+      Qt.openUrlExternally("tel: ///"+callUrl);
+      request.dialogAccept();
 
-      } else if(request.message.toLowerCase().includes("http")){
-          Qt.openUrlExternally(request.message);
-          request.dialogAccept();
-      } else if(request.message =="paste"){
-        request.dialogAccept(UITK.Clipboard.data.text ? UITK.Clipboard.data.text : "");
-  } else{
+    }
+    else if(request.message.toLowerCase().includes("http"))
+    {
+      Qt.openUrlExternally(request.message);
+      request.dialogAccept();
+    }
+    else if(request.message =="paste")
+    {
+      request.dialogAccept(UITK.Clipboard.data.text ? UITK.Clipboard.data.text: "");
+    }
+    else {
       simpleDialog.request = request;
       simpleDialog.visible = true;
       root.request = request;
-
     }
-
+    onFeaturePermissionRequested: {
+      grantFeaturePermission(securityOrigin, feature, true);
     }
```

Comment:
Where are those variables defined?
And does this ask for permission for any feature the vue code asks for? Seems like a security risk.

Reply:
This asks for permission on the lomiri level. We could add a dialog here for non Lomiri qt environements but i think that's not necessary.

Reply:
Can we restrict for which features we potentially ask for permission?

If some hacker is taking over our frontend, and this code allows him to ask for any right he wants, this may be his door to take over the whole phone.